src/vm/eventpipe.cpp

```diff
@@ -166,7 +166,12 @@ EventPipeSessionID EventPipe::Enable(
         if (!s_tracingInitialized)
             return;
 
-        EventPipeSession *const pSession = s_config.CreateSession(
+        const uint32_t SessionIndex = GenerateSessionIndex();
+        if (SessionIndex >= EventPipe::MaxNumberOfSessions)
+            return;
+
+        EventPipeSession *const pSession = new EventPipeSession(
```

Comment:
What if we run out of memory here?

Reply:
if we run out-of-memory, then it will throw (std::bad_alloc?).